diff --git a/hw-balancedparens.cabal b/hw-balancedparens.cabal
index 56cfc70..eec8ba3 100644
--- a/hw-balancedparens.cabal
+++ b/hw-balancedparens.cabal
@@ -75,57 +75,3 @@ library
   autogen-modules:    Paths_hw_balancedparens
   hs-source-dirs:     src
 
-library hw-balancedparens-gen
-  import:   base, config
-          , deepseq
-          , hedgehog
-          , hspec
-          , hw-prim
-          , vector
-  exposed-modules:
-    HaskellWorks.Data.BalancedParens.Gen
-    Paths_hw_balancedparens
-  build-depends:      hw-balancedparens
-  hs-source-dirs:     gen
-  autogen-modules:    Paths_hw_balancedparens
-
-test-suite hw-balancedparens-test
-  import:   base, config
-          , hedgehog
-          , hspec
-          , hw-bits
-          , hw-hspec-hedgehog
-          , hw-prim
-          , hw-rankselect-base
-          , transformers
-          , vector
-  type:               exitcode-stdio-1.0
-  main-is:            Spec.hs
-  other-modules:
-    HaskellWorks.Data.BalancedParens.Internal.BroadwordSpec
-    HaskellWorks.Data.BalancedParens.Internal.ParensSeqSpec
-    HaskellWorks.Data.BalancedParens.RangeMin2Spec
-    HaskellWorks.Data.BalancedParens.RangeMinSpec
-    HaskellWorks.Data.BalancedParens.SimpleSpec
-    Paths_hw_balancedparens
-  build-depends:      hw-balancedparens
-                    , hw-balancedparens-gen
-  hs-source-dirs:     test
-  ghc-options:        -threaded -rtsopts -with-rtsopts=-N
-  autogen-modules:    Paths_hw_balancedparens
-  build-tool-depends: hspec-discover:hspec-discover
-
-benchmark bench
-  import:   base, config
-          , criterion
-          , hedgehog
-          , hw-bits
-          , hw-prim
-          , vector
-  type:               exitcode-stdio-1.0
-  main-is:            Main.hs
-  other-modules:      Paths_hw_balancedparens
-  autogen-modules:    Paths_hw_balancedparens
-  hs-source-dirs:     bench
-  build-depends:      hw-balancedparens
-                    , hw-balancedparens-gen
